pkg/apis/seedmanagement/types_managedseed.go

```diff
@@ -0,0 +1,161 @@
+// the image, etc.
+type GardenletDeployment struct {
+	// ReplicaCount is the number of gardenlet replicas. Defaults to 1.
+	ReplicaCount *int32
+	// RevisionHistoryLimit is the number of old gardenlet ReplicaSets to retain to allow rollback. Defaults to 1.
+	RevisionHistoryLimit *int32
+	// ServiceAccountName is the name of the ServiceAccount to use to run gardenlet pods.
+	ServiceAccountName *string
+	// Image is the gardenlet container image.
+	Image *Image
+	// Resources are the compute resources required by the gardenlet container.
+	Resources *corev1.ResourceRequirements
+	// PodLabels are the labels on gardenlet pods.
+	PodLabels map[string]string
+	// PodAnnotations are the annotations on gardenlet pods.
+	PodAnnotations map[string]string
+	// AdditionalVolumes is the list of additional volumes that should be mounted by gardenlet containers.
+	AdditionalVolumes []corev1.Volume
+	// AdditionalVolumeMounts is the list of additional pod volumes to mount into the gardenlet container's filesystem.
+	AdditionalVolumeMounts []corev1.VolumeMount
+	// Env is the list of environment variables to set in the gardenlet container.
+	Env []corev1.EnvVar
+	// VPA specifies whether to enable VPA for gardenlet. Defaults to true.
+	VPA *bool
+}
+
+// Image specifies container image parameters.
+type Image struct {
+	// Repository is the image repository.
+	Repository *string
+	// Tag is the image tag.
+	Tag *string
+	// PullPolicy is the image pull policy. One of Always, Never, IfNotPresent.
+	// Defaults to Always if latest tag is specified, or IfNotPresent otherwise.
+	PullPolicy *corev1.PullPolicy
+}
+
+// Bootstrap describes a mechanism for bootstrapping gardenlet connection to the Garden cluster.
+type Bootstrap string
+
+const (
+	// BootstrapServiceAccount means that a temporary service account should be used for bootstrapping gardenlet connection to the Garden cluster.
+	BootstrapServiceAccount Bootstrap = "ServiceAccount"
+	// BootstrapToken means that a bootstrap token should be used for bootstrapping gardenlet connection to the Garden cluster.
+	BootstrapToken Bootstrap = "BootstrapToken"
+	// BootstrapNone means that gardenlet connection to the Garden cluster should not be bootstrapped
+	// and the gardenClientConnection.kubeconfig field should be used to connect to the Garden cluster.
+	BootstrapNone Bootstrap = "None"
+)
+
+// ManagedSeedStatus is the status of a ManagedSeed.
+type ManagedSeedStatus struct {
+	// Conditions represents the latest available observations of a ManagedSeed's current state.
+	Conditions []gardencore.Condition
+	// ObservedGeneration is the most recent generation observed for this ManagedSeed. It corresponds to the
+	// ManagedSeed's generation, which is updated on mutation by the API Server.
+	ObservedGeneration int64
+}
+
+const (
+	// ManagedSeedShootExists is a condition type for indicating whether the ManagedSeed's shoot exists.
+	ManagedSeedShootExists gardencore.ConditionType = "ShootExists"
```

Comment:
I still believe this is needed. If for whatever reason the shoot is not found when reconciling the managed seed (e.g. admission plugins were inactive or buggy), shouldn't this situation be reflected in the status, similarly to all other errors? If yes, then none of the other conditions would do since this case is really special, also it would fail earlier. If you are fine with keeping it, let's keeping and decide later during the review of the controller implementation.

Reply:
Hm, I'm not aware of the details of the controller implementation yet, but the argument ("If for whatever reason the shoot is not found when reconciling the managed seed (e.g. admission plugins were inactive or buggy)") sounds artificial/unlikely. If that's the only point then let's drop it and rather populate such issues with other conditions or events as @timebertt suggested. Personally, I'm fine with keeping it for now and revisiting it in the respective controller PR.

Reply:
OK, let's keep it then, will remove it later if needed.